// @rnmapbox/maps/lib/module/components/NativeUserLocation.js
import React, { memo } from 'react';
import { processColor } from 'react-native';
import RNMBXNativeUserLocation from '../specs/RNMBXNativeUserLocationNativeComponent';
const defaultProps = {
  visible: true
};
const NativeUserLocation = /*#__PURE__*/memo(props => {
  const {
    iosShowsUserHeadingIndicator,
    pulsing,
    ...rest
  } = props;
  const nativePulsing = pulsing ? _pulsingToNative(pulsing) : undefined;
  let baseProps = {
    ...defaultProps,
    pulsing: nativePulsing
  };
  if (iosShowsUserHeadingIndicator) {
    console.warn('NativeUserLocation: iosShowsUserHeadingIndicator is deprecated, use puckBearingEnabled={true} puckBearing="heading" instead');
    baseProps = {
      ...baseProps,
      puckBearingEnabled: true,
      puckBearing: 'heading'
    };
  }
  const actualProps = {
    ...baseProps,
    ...rest
  };
  return /*#__PURE__*/React.createElement(RNMBXNativeUserLocation, actualProps);
});
function _pulsingToNative(pulsing) {
  if (pulsing === 'default') {
    return {
      kind: 'default'
    };
  }
  if (pulsing == null) {
    return undefined;
  }
  const {
    color,
    isEnabled,
    radius
  } = pulsing;
  return {
    color: processColor(color),
    isEnabled,
    radius
  };
}
export default NativeUserLocation;
