// @rnmapbox/maps/lib/commonjs/RNMBXModule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.setWellKnownTileServer = exports.setTelemetryEnabled = exports.setConnected = exports.setAccessToken = exports.removeCustomHeader = exports.getAccessToken = exports.addCustomHeader = exports.TileServers = exports.StyleURL = exports.StyleSource = exports.OfflinePackDownloadState = exports.LineJoin = void 0;
var _reactNative = require("react-native");
const RNMBXModule = {
  ..._reactNative.NativeModules.RNMBXModule
};
const {
  StyleURL,
  OfflinePackDownloadState,
  LineJoin,
  StyleSource,
  TileServers,
  removeCustomHeader,
  addCustomHeader,
  setAccessToken,
  setWellKnownTileServer,
  getAccessToken,
  setTelemetryEnabled,
  setConnected
} = RNMBXModule;
exports.setConnected = setConnected;
exports.setTelemetryEnabled = setTelemetryEnabled;
exports.getAccessToken = getAccessToken;
exports.setWellKnownTileServer = setWellKnownTileServer;
exports.setAccessToken = setAccessToken;
exports.addCustomHeader = addCustomHeader;
exports.removeCustomHeader = removeCustomHeader;
exports.TileServers = TileServers;
exports.StyleSource = StyleSource;
exports.LineJoin = LineJoin;
exports.OfflinePackDownloadState = OfflinePackDownloadState;
exports.StyleURL = StyleURL;
